src/configs/types.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod';
+
+export const zConfigSchema = z.object({
+  id: z.string(),
+  version: z.string(),
+  value: z.record(z.string(), z.unknown()),
+});
+
+export type Config = z.infer<typeof zConfigSchema>;
+
+export enum ConfigSpecialVersion {
+  LATEST = 'latest',
+  DANGEROUSLY_USE_UNDEPLOYED = 'dangerously-use-undeployed',
+}
+
+export type ConfigVersion = ConfigSpecialVersion | string;
+
+export const zConfigParameterSchema = z.union([
+  z.object({
+    id: z.string(),
+    latest: z.literal(true),
+  }),
+  z.object({
+    id: z.string(),
+    dangerouslyUseUndeployed: z.literal(true),
+  }),
+  z.object({
+    id: z.string(),
+    version: z.string(),
+  }),
+  z.object({
+    id: z.string(),
+    revisionId: z.string(),
```

Comment:
i think we want pinning to a specific revisionId under the dangerously-use-undeployed umbrella, since you'd only want to use a specific revision during local development and the dangerously-use-* prefix is intended to be a clue to you (and to code reviewers if you accidentally open a pull request w/o backing out your temporary changes) that it's not intended to be used in production. as-is you would be able to do:

```ts
await config.activateRemoteConfig({
  config: {
    id: 'my-config-id',
    revisionId: 'clvifur81000336hksycpw4av',
  },
});
```

which doesn't have any hints that you're about to commit something dangerous (an undeployed revision) to production

so for prompts for example, my plan is to update the manager to be like:

```ts
const mgr = new AutoblocksPromptManager({
  id: 'my-prompt',
  version: {
    // Use latest undeployed revision; requires a user-scoped API key
    major: 'dangerously-use-undeployed',
    minor: 'latest',
  },
});

const mgr = new AutoblocksPromptManager({
  id: 'my-prompt',
  version: {
    // Use specific undeployed revision; requires a user-scoped API key
    major: 'dangerously-use-undeployed',
    minor: 'clvifur81000336hksycpw4av',
  },
});

const mgr = new AutoblocksPromptManager({
  id: 'my-prompt',
  version: {
    // Use deployed revision 1.0
    major: '1',
    minor: '0',
  },
});

const mgr = new AutoblocksPromptManager({
  id: 'my-prompt',
  version: {
    // Use latest deployed revision w/ major version 1
    major: '1',
    minor: 'latest',
  },
});
```

in other words, there is no way to specify a specific revisionId w/o having `dangerously-use-undeployed` there to hint that you're using something dangerous intended only for local dev scenarios

Reply:
i wonder if it should be like:

```ts
export const zConfigParameterSchema = z.union([
  ...
  z.object({
    id: z.string(),
    // Using this requires a user-scoped API key
    dangerouslyUseUndeployed: z.union([
      z.object({
        // Use latest undeployed revision
        latest: z.literal(true),
      }),
      z.object({
        // Use specific undeployed revision
        revisionId: z.string(),
      }),
    ]),
  }),
  ...
]);
```

?

Reply:
Ya I think that makes sense. I can update to that format